openapi-cli/src/main/java/io/ballerina/openapi/generators/client/mime/UrlEncodedType.java

```diff
@@ -0,0 +1,122 @@
+import io.swagger.v3.oas.models.media.MediaType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createEmptyNodeList;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createIdentifierToken;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createSeparatedNodeList;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createToken;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createBuiltinSimpleNameReferenceNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createCaptureBindingPatternNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createMappingConstructorExpressionNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createTypedBindingPatternNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createVariableDeclarationNode;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.CLOSE_BRACE_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.EQUAL_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.OPEN_BRACE_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.SEMICOLON_TOKEN;
+import static io.ballerina.openapi.generators.GeneratorConstants.ENCODING;
+import static io.ballerina.openapi.generators.GeneratorConstants.STRING;
+import static io.ballerina.openapi.generators.GeneratorUtils.createEncodingMap;
+
+/**
+ * Defines the payload structure of "application/x-www-form-urlencoded" mime type.
+ */
+public class UrlEncodedType extends MimeType {
+    private final BallerinaUtilGenerator ballerinaUtilGenerator;
+
+    public UrlEncodedType(BallerinaUtilGenerator ballerinaUtilGenerator) {
+        this.ballerinaUtilGenerator = ballerinaUtilGenerator;
+    }
+
+    @Override
+    public void setPayload(List<StatementNode> statementsList, Map.Entry<String, MediaType> mediaTypeEntry) {
+        ballerinaUtilGenerator.setRequestBodyEncodingFound(true);
+        VariableDeclarationNode requestBodyEncodingMap = getRequestBodyEncodingMap(
+                mediaTypeEntry.getValue().getEncoding());
+        payloadName = "encodedRequestBody";
+        if (requestBodyEncodingMap != null) {
+            statementsList.add(requestBodyEncodingMap);
+            VariableDeclarationNode requestBodyVariable = generatorUtils.getSimpleStatement(STRING,
+                    payloadName,
+                    "createFormURLEncodedRequestBody(payload, requestBodyEncoding)");
+            statementsList.add(requestBodyVariable);
+        } else {
+            VariableDeclarationNode requestBodyVariable = generatorUtils.getSimpleStatement(STRING,
+                    payloadName, "createFormURLEncodedRequestBody(payload)");
+            statementsList.add(requestBodyVariable);
+        }
+        setPayload(statementsList, payloadName, mediaTypeEntry.getKey());
+    }
+
+    /**
```

Comment:
Shall we add a test for "application/x-www-form-urlencoded" scenarios?

Reply:
I guess it is already added. https://github.com/ballerina-platform/ballerina-openapi/blob/5eea4bc562af252954a8318ce7e8c1e74f9047f7/openapi-cli/src/test/java/io/ballerina/openapi/generators/client/UtilGenerationTests.java#L95
@aneeshafedo Can you confirm?

Reply:
I have added tests to check the util generations for `application/x-www-form-urlencoded`. However it will be better if we can add another under `RequestBodyTests`. Shall we add it in this PR or shall we addressed in a seperate PR ?

Reply:
Added in https://github.com/ballerina-platform/ballerina-openapi/commit/fa0e11da767a053ad12e56dfc381c476d07349e2